docs/policy-only.md

```diff
@@ -0,0 +1,156 @@
+host route with corresponding Pod's IP address as destination is created on each PtP device. Within
+each Pod, routes are configured to ensure all outgoing traffic is sent over this PtP device, and
+incoming traffic is received on this PtP device. This is a spoke-and-hub model, where to/from Pod
+traffic, even within the same worker Node must traverse first to the host network and being
+routed by it.
+
+When a Pod is instantiated, the container runtime first calls the primary CNI to configure
+Pod's IP, route table, DNS etc, and then connects Pod to host network with a PtP device such as a
+veth-pair. When CNI chaining is configured, container runtime then calls Antrea, and the Antrea
+agent attaches Pod's PtP device to the OVS bridge, and moves the host route to the Pod to local
+host gateway(``gw0``) interface from the PtP device. This is illustrated by the diagram on the
+right.
+
+Antrea needs to satisfy that 
+1. All IP packets, sent on ``gw0`` in the host network, are received by the Pods exactly the same
+as if the OVS bridge has not been inserted. 
+1. Similarly all IP packets, sent by Pods, are received by other Pods or the host network exactly
+the same as if OVS bridge has not been inserted.
+1. There are no requirements on Pod MAC addresses as all MAC addresses stays within the OVS bridge.
+
+To satisfy the above requirements, Antrea needs no knowledge of Pod's network configurations nor
+nor underlying CNI network, it simply needs to program the following OVS flows on the OVS bridge:
+1. A default ARP responder flow that answers any ARP request. Its sole purpose is so that a Pod's
+neighbor may be resolved, and packets may be sent by that Pod to that neighbor.
+1. IP packets are routed based on its destination IP if it matches any local Pod's IP.
+1. All other IP packets are routed to host network via gw0 interface.
+
+These flows together handle all Pod traffic patterns with exception of Pod-to-Service traffic
+that we will address next.
+
+## Handling Pod-To-Service
+The discussion in this section is relevant also to Pod-to-Service traffic in NoEncap traffic
+mode. Antrea applies same principle to handle Pod-to-service traffic in all traffic modes where
+traffic requires no encapsulation.
+
+Antrea uses kube-proxy for load balancing. At the same time, it supports also Pod level
+NetworkPolicy enforcement.
+
+This means that for any Pod-to-Service traffic flow, it needs to  
+1. first traverse to the host network for load balancing (DNAT), then
+1. comes back to OVS bridge for Pod NetworkPolicy processing, and
+1. if DNATed destination in 1) is an inter-Node Pod or external network entity, and goes back
+to the host network yet again to be forwarded.
+
+We refer the last traffic pattern as re-entrance traffic because in this pattern, an traffic flow
+enters host network twice -- first time for load balancing, and second time for forwarding.
+
```

Comment:
At this point it seems everything I have read in this section also applies to encap mode?

Reply:
the difference is that service traffic is encap mode does not re-enter host network with same 5-tuple.